plan.go

```diff
@@ -115,21 +116,32 @@ func createSteps(ctx *PlanningContext, insertionPoint []string, parentType, pare
 	return result, nil
 }
 
-func extractSelectionSet(ctx *PlanningContext, insertionPoint []string, parentType string, input ast.SelectionSet, location string, childstep bool) (ast.SelectionSet, []*QueryPlanStep, error) {
+var reservedAliases = map[string]string{
+	"__typename": "__typename",
+	"_bramble_id": "id",
+}
+
+func extractSelectionSet(ctx *PlanningContext, insertionPoint []string, parentType string, input ast.SelectionSet, location string) (ast.SelectionSet, []*QueryPlanStep, error) {
 	var selectionSetResult []ast.Selection
 	var childrenStepsResult []*QueryPlanStep
 	var remoteSelections []ast.Selection
 	for _, selection := range input {
 		switch selection := selection.(type) {
 		case *ast.Field:
+			for reservedAlias, requiredName := range reservedAliases {
+				if selection.Alias == reservedAlias && selection.Name != requiredName {
+					return nil, nil, gqlerror.Errorf("%s.%s: alias \"%s\" is reserved for system use", strings.Join(insertionPoint, "."), reservedAlias, reservedAlias)
+				}
+			}
```

Comment:
This loop and the `reservedAliases` struct above block the submission of reserved aliases. If the user attempts to make a selection that will interfere with internal Bramble operations, we'll simply tell them it's not allowed.

Reply:
Hey I was curious about this part here so did some testing - I was mainly wondering if this would effect `__typename` fields in selection sets. Seems to be fine, moreover, it's fine with `__typename: __typename` which I thought was fishy. Then I tried out `_bramble_id: id`, this also works without error. Perhaps this isn't working as intended?

Here's a test (for `plan_test.go`):
```go
func TestQueryBrambleReserved(t *testing.T) {
	query := `{
		movies {
			... on Movie {
				id
				title(language: French)
				_bramble_id: id
			}
		}
	}`
	plan := `{
		"RootSteps": [
			{
				"ServiceURL": "A",
				"ParentType": "Query",
        "SelectionSet": "{ movies { ... on Movie { id title(language: French) _bramble_id: id } _bramble_id: id } }",
				"InsertionPoint": null,
				"Then": null
			}
		]
	}`
	PlanTestFixture1.Check(t, query, plan) // we actually expect a panic here, but looking at the diff here is useful
}
```

Reply:
Ah nevermind this, didn't quite see the last part of the condition being a `!=`. Sorry for the noise.